domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2379,26 +2379,42 @@ private FacetManager buildFacetManager(CQLWrapper wrapper, QueryHelper queryHelp
 
   /**
    * Get the jsonb by id.
+   * @param conn  if provided, the connection on which to execute the query on.
+   * @param lock  whether to use SELECT FOR UPDATE to lock the selected row.
    * @param table  the table to search in
    * @param id  the value of the id field
    * @param function  how to convert the (String encoded) JSON
    * @param replyHandler  the result after applying function
    */
-  private <R> void getById(String table, String id,
+  private <R> void getById(final AsyncResult<SQLConnection> conn, boolean lock, String table, String id,
                            FunctionWithException<String, R, Exception> function,
                            Handler<AsyncResult<R>> replyHandler) {
-    getConnection(res -> {
-      if (res.failed()) {
-        replyHandler.handle(Future.failedFuture(res.cause()));
-        return;
+    Promise<PgConnection> promise = Promise.promise();
+    if (conn != null) {
+      if (conn.failed()) {
+        promise.fail(conn.cause());
+      } else {
+        promise.complete(conn.result().conn);
       }
-      PgConnection connection = res.result();
+    } else {
+      getConnection(res -> {
+        if (res.failed()) {
+          promise.fail(res.cause());
+        } else {
+          promise.complete(res.result());
+        }
+      });
+    }
+    promise.future().onFailure(ex -> replyHandler.handle(Future.failedFuture(ex)))
+    .onSuccess(connection -> {
       String sql = SELECT + DEFAULT_JSONB_FIELD_NAME
           + FROM + schemaName + DOT + table
           + WHERE + ID_FIELD + "= $1";
+      if (lock) {
+        sql += " FOR UPDATE";
+      }
```

Comment:
Can you write a test with a real PostgreSQL (not a mock) to ensure that "FOR UPDATE" actually works:
On first connection run "SELECT ... FOR UPDATE".
On second connection run "SELECT" for that row, this will block until the first connection closes the transaction.
On first connection run "SELECT pg_sleep(0.1)", "UPDATE" the record, and COMMIT the transaction.
The second connection should return the new value from the UPDATE, not the old value.